server/controllers/mensa.ts

```diff
@@ -84,19 +85,75 @@ router.get('/', async (req, res, next) => {
       return result.sort((a, b) => moment(a.date).isBefore(moment(b.date)) ? -1 : 1).slice(0, amountOfReturnedDays);
     }, { ttl: CACHE_SECONDS });
 
+    return data;
+  } catch (error) {
+    console.log('Error');
+  }
+}
+
+router.get('/', async (req, res, next) => {
+  let mensaList:Mensa[] = [];
+  const mensaIDs = [130, 112, 200, 134, 132]; // list of ostfalia mensas
+  const mensaPromises = mensaIDs.map(async mensaID => {
+    const key = 'mensa-' + mensaID;
+    try {
+      const data = await cache.wrap(key, async () => {
+        console.log(`mensa cache miss for key ${key}`);
+
+        const dayPlans = await getDayPlan(mensaID);
+
+        const signal = timeoutSignal(3000) // abort if openmensa is too slow to respond
+
+        const response = await fetch(`https://sls.api.stw-on.de/v1/location/${mensaID}`, { signal }) // 130 is WF mensa
+          .then((res) => res.json());
+
+        let url;
+        switch (response.id) {
+          case 112:
+            url = 'bistro4u';
+            break;
+
+          default:
+            url = 'mensa';
+            break;
+        }
+
+        const mensa:Mensa = {
+          // copy only neccessary fields
+          name: response.name,
+          id: response.id,
+          dayPlans,
+          url: `${response.address.city.toLowerCase()}/essen/${url}`,
+          opening_hours: response.opening_hours/* .map(({ ['time']: _, ...rest }) => rest) */.filter((obj, index, self) =>
```

Comment:
Es scheitn uebrigens leider so, als waren die zurueckgegebenen Oeffnungszeiten falsch, zumindest fuer Salzgitter wird in der API sonntag zuruckgegeben, aber die realen zeiten sind laut website diese:

![image](https://github.com/user-attachments/assets/df3d5581-01b2-49d6-ae6b-eb0eddb42dc5)

Reply:
ich habe denen mal geschrieben. Ich hoffe die können das soweit korrigieren, eigentlich komisch, dass deren Seite nicht die API dafür nutzt... ich find es eigentlich ganz cool. Was hälst du davon Einträge am Wochenende einfach zu entfernen, dann würde es zumindest grob stimmen

Reply:
Klingt gut!